actors/market/tests/market_actor_test.rs

```diff
@@ -953,6 +953,145 @@ fn do_not_terminate_deal_if_end_epoch_is_equal_to_or_less_than_current_epoch() {
     assert_deals_not_terminated(&mut rt, &[deal2]);
 }
 
+// Converted from: https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/market/market_test.go#L1436
+#[test]
+fn fail_when_caller_is_not_a_storage_miner_actor() {
+    let provider_addr = Address::new_id(PROVIDER_ID);
+
+    let mut rt = setup();
+    rt.expect_validate_caller_type(vec![*MINER_ACTOR_CODE_ID]);
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, provider_addr);
+    let params = OnMinerSectorsTerminateParams { epoch: rt.epoch, deal_ids: vec![] };
+
+    // XXX: Which exit code is correct: SYS_FORBIDDEN(8) or USR_FORBIDDEN(18)?
+    assert_eq!(
+        ExitCode::USR_FORBIDDEN,
+        rt.call::<MarketActor>(
+            Method::OnMinerSectorsTerminate as u64,
+            &RawBytes::serialize(params).unwrap(),
+        )
+        .unwrap_err()
+        .exit_code()
+    );
+}
+
+// Converted from: https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/market/market_test.go#L1448
+#[test]
+fn fail_when_caller_is_not_the_provider_of_the_deal() {
+    let start_epoch = 10;
+    let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
+    let sector_expiry = end_epoch + 100;
+    let current_epoch = 5;
+    let owner_addr = Address::new_id(OWNER_ID);
+    let provider_addr = Address::new_id(PROVIDER_ID);
+    let worker_addr = Address::new_id(WORKER_ID);
+    let client_addr = Address::new_id(CLIENT_ID);
+    let control_addr = Address::new_id(CONTROL_ID);
+
+    let provider2 = Address::new_id(501);
+
+    let mut rt = setup();
+    rt.set_epoch(current_epoch);
+
+    let deal = generate_and_publish_deal(
+        &mut rt,
+        client_addr,
+        provider_addr,
+        owner_addr,
+        worker_addr,
+        control_addr,
+        start_epoch,
+        end_epoch,
+    );
+    activate_deals(&mut rt, sector_expiry, provider_addr, current_epoch, &[deal]);
+
+    // XXX: Difference between go messages: 't0501' has turned into 'f0501'.
```

Comment:
This is probably not a serious issue because test net and main net addresses are interchangeable apart from how they display when printed out.  Before merging please file an issue in builtin actors so we don't lose track of this problem.

Reply:
Done. #272